src/Microsoft.AspNetCore.Sockets.Abstractions/Features/IConnectionUserFeature.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Sockets.Features
+{
+    public interface IConnectionUserFeature
+    {
+        ClaimsPrincipal User { get; }
```

Comment:
Why is this **not** settable? Of all the feature properties, it's the one middleware is most likely to want to mutate.